File: app/src/main/java/com/tomaschlapek/tcbasearchitecture/widget/NonSwipeableViewPager.java
package com.tomaschlapek.tcbasearchitecture.widget;

import android.content.Context;
import android.util.AttributeSet;
import android.view.MotionEvent;

import android.support.v4.view.ViewPager;

/**
 * Created by tomaschlapek on 15/2/17.
 */

public class NonSwipeableViewPager extends ViewPager {

  private boolean mSwipeable;

  /* Public Methods *******************************************************************************/

  /**
   * @see ViewPager#ViewPager(Context)
   */
  public NonSwipeableViewPager(Context context) {
    super(context);
  }

  /**
   * @see ViewPager#ViewPager(Context, AttributeSet)
   */
  public NonSwipeableViewPager(Context context, AttributeSet attrs) {
    super(context, attrs);
  }

  //////////////////////////////////////////////////////////////////////////////////////////////////

  /**
   * @see ViewPager#onInterceptTouchEvent(MotionEvent)
   */
  @Override
  public boolean onInterceptTouchEvent(MotionEvent event) {
    // Never allow swiping to switch between pages.
    return mSwipeable;
  }

  /**
   * @see ViewPager#onTouchEvent(MotionEvent)
   */
  @Override
  public boolean onTouchEvent(MotionEvent event) {
    // Never allow swiping to switch between pages.
    return mSwipeable;
  }

  public boolean isSwipeable() {
    return mSwipeable;
  }

  public void setSwipeable(boolean swipeable) {
    mSwipeable = swipeable;
  }

}

File: app/src/main/kotlin/com/tomaschlapek/tcbasearchitecture/widget/NonSwipeableViewPager.kt
package com.tomaschlapek.tcbasearchitecture.widget

import android.content.Context
import android.support.v4.view.ViewPager
import android.util.AttributeSet
import android.view.MotionEvent

/**
 * view pager with disabled swiping.
 */
class NonSwipeableViewPager : ViewPager {

  var isSwipeable: Boolean = false

  /* Public Methods *******************************************************************************/

  /**
   * @see ViewPager.ViewPager
   */
  constructor(context: Context) : super(context) {}

  /**
   * @see ViewPager.ViewPager
   */
  constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

  //////////////////////////////////////////////////////////////////////////////////////////////////

  /**
   * @see ViewPager.onInterceptTouchEvent
   */
  override fun onInterceptTouchEvent(event: MotionEvent): Boolean {
    // Never allow swiping to switch between pages.
    return isSwipeable
  }

  /**
   * @see ViewPager.onTouchEvent
   */
  override fun onTouchEvent(event: MotionEvent): Boolean {
    // Never allow swiping to switch between pages.
    return isSwipeable
  }

}